pkg/node/sendoperation/deploySC/deploySC.go

```diff
@@ -0,0 +1,44 @@
+package deploysc
+
+import (
+	"io"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/massalabs/thyra/api/swagger/server/models"
+	"github.com/massalabs/thyra/api/swagger/server/restapi/operations"
+	"github.com/massalabs/thyra/pkg/node"
+	"github.com/massalabs/thyra/pkg/onchain"
+)
+
+func Handler(params operations.CmdDeploySCParams) middleware.Responder {
+	client := node.NewDefaultClient()
+
+	file, err := io.ReadAll(params.Wasmfile)
+	if err != nil {
+		return operations.NewCmdDeploySCBadRequest().
+			WithPayload(
+				&models.Error{
+					Code:    err.Error(),
+					Message: err.Error(),
+				})
+	}
+
+	address, err := onchain.DeploySCV2(client,
+		params.Nickname,
+		*params.GazLimit,
```

Comment:
GazLimist now a pointer which can be undefined. The code will panic if it's undefined here.

You have three options:
- add `x-nullable : false` in the swagger spec, but it's incompatible with default value if I remember well
- check before using it that the pointer is not nil
- verify that go-swagger is doing the job properly and add a comment expaning that everything is fine

Reply:
Go-swagger is working properly :)